google/cloud/pubsub/samples/samples.cc

```diff
@@ -1024,6 +1024,73 @@ void SubscriberRetrySettings(std::vector<std::string> const& argv) {
       std::move(p.second), __func__);
 }
 
+void OptimisticSubscribe(std::vector<std::string> const& argv) {
+  namespace examples = ::google::cloud::testing_util;
+  if (argv.size() != 3) {
+    throw examples::Usage{
+        "optimistic-subscribe <project-id> <topic-id> <subscription-id>"};
+  }
+  namespace pubsub_admin = ::google::cloud::pubsub_admin;
+  namespace pubsub = ::google::cloud::pubsub;
+  [](std::string project_id, std::string topic_id,
+     std::string subscription_id) {
+    // Do not retry the attempts to consume messages.
+    auto subscriber = pubsub::Subscriber(
+        pubsub::MakeSubscriberConnection(
+            pubsub::Subscription(project_id, subscription_id)),
+        google::cloud::Options{}.set<pubsub::RetryPolicyOption>(
+            pubsub::LimitedErrorCountRetryPolicy(
+                /*maximum_failures=*/0)
+                .clone()));
+
+    // [START pubsub_optimistic_subscribe]
+    try {
+      // Try to consume from subscription.
+      auto response = subscriber.Pull();
+      if (!response) {
+        if (response.status().code() ==
+            google::cloud::StatusCode::kUnavailable) {
+          std::cout << "Pull received kUnavailable status: "
```

Comment:
Why is unavailable a special error?

Reply:
It looks like we actually create the error code. I updated it to check for the error message too.